src/compiler/diagnosticMessages.json

```diff
@@ -1252,6 +1252,10 @@
         "category": "Error",
         "code": 2366
     },
+    "The types of these values indicate that this condition will always be '{0}'.": {

```

Comment:
If those are statically known primitive types, can we include actual values in error message? Like:

`"These values are known to be '{0}' and '{1}', rendering this condition always '{2}'."`

Reply:
Check out #25311